## 86% of end-end tests have passed
```
Total executed tests: 7
Failed tests: 1
Expected to fail tests: 0
Passed tests: 6
```
```
IDs of failed tests: 702745 
```
<h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=fix/status-bar-bug&PR_ID=21105&APK_NAME=21105.apk&TR_CASE_IDS=702745">Rerun failed tests</a></li><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3643731">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div>Device 1: Find `Button` by `accessibility id`: `Crop` </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:278: in test_1_1_chat_non_latin_messages_stack_update_profile_photo
    self.profile_1.edit_profile_picture(image_index=2)
../views/profile_view.py:459: in edit_profile_picture
    self.crop_photo_button.click()
../views/base_element.py:90: in click
    element = self.find_element()
../views/base_element.py:84: in find_element
    raise exception
../views/base_element.py:77: in find_element
    return self.driver.find_element(self.by, self.locator)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:409: in find_element
    return self.execute(RemoteCommand.FIND_ELEMENT, {'using': by, 'value': value})['value']
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E       at UIA2Proxy.command (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/jsonwp-proxy/proxy.js:355:13)
E       at processTicksAndRejections (node:internal/process/task_queues:95:5)
E       at AndroidUiautomator2Driver.helpers.doFindElementOrEls (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/lib/commands/find.js:30:10)
E       at doFind (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/node_modules/appium-android-driver/lib/commands/find.js:36:17)
E       at wrappedCondFn (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/timeout.ts:137:14)
E       at spin (/mnt/sauce/appium/appium-v2.0.0/node_modules/asyncbox/lib/asyncbox.js:221:20)
E       at waitForCondition (/mnt/sauce/appium/appium-v2.0.0/node_modules/asyncbox/lib/asyncbox.js:238:10)
E       at AndroidUiautomator2Driver.implicitWaitForCondition (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/timeout.ts:139:12)
E       at AndroidUiautomator2Driver.findElOrEls (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/node_modules/appium-android-driver/lib/commands/find.js:60:5)
E       at AndroidUiautomator2Driver.findElOrElsWithProcessing (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/find.ts:60:12)
E       at AndroidUiautomator2Driver.findElement (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/find.ts:75:12)
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d45d9adee13b4e289dc0be76d97f5057?auth=ffbfbe934ce0234c54bdd46b8ed15ec6#105">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e873dc5f45b8408eb3223503138ebd6b?auth=67ac11ac862746a65f6bca11ba9808d0#90">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (6)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3643733">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0e3f68a6409140ec8f3e4ac8de9da8ed?auth=6cbe819c3b786b5e531f2e2342deb98d#82">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3643730">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0e3f68a6409140ec8f3e4ac8de9da8ed?auth=6cbe819c3b786b5e531f2e2342deb98d#53">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3643735">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3643734">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3643736">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d57ddbfd1a44473590881c1f693af597?auth=44c72a3084c1760ebd44e50a67b6584f#39">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3643732">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/dd12af0d9e7249f38565dcb29575763d?auth=c232599924d4dfd1b85b071167371a9e#168">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e0ad450e7717430887526b84ef5f6844?auth=ebab056c6d9f8b2db22e6bb55e670626#107">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/legacy/status_im/multiaccounts/logout/core.cljs
+++ b/src/legacy/status_im/multiaccounts/logout/core.cljs
@@ -30,7 +30,7 @@
   [{:keys [db] :as cofx} {:keys [auth-method logout?]}]
   (let [key-uid (get-in db [:profile/profile :key-uid])]
     (rf/merge cofx
-              {:dispatch                               [:init-root :progress]
+              {:dispatch                               [:update-theme-and-init-root :progress]
                :effects.shell/reset-state              nil
                :hide-popover                           nil
                ::logout                                nil

--- a/src/status_im/contexts/onboarding/events.cljs
+++ b/src/status_im/contexts/onboarding/events.cljs
@@ -91,7 +91,7 @@
     (merge
      {:db (assoc db :profile/profiles-overview multiaccounts)}
      (when-not (seq multiaccounts)
-       {:dispatch [:init-root :screen/onboarding.intro]}))))
+       {:dispatch [:update-theme-and-init-root :screen/onboarding.intro]}))))
 
 (rf/defn password-set
   {:events [:onboarding/password-set]}

--- a/src/status_im/contexts/onboarding/welcome/view.cljs
+++ b/src/status_im/contexts/onboarding/welcome/view.cljs
@@ -45,7 +45,7 @@
           :colors [colors/neutral-100-opa-0 colors/neutral-100-opa-80]}])
       [quo/button
        {:on-press            (fn []
-                               (rf/dispatch [:init-root :shell-stack])
+                               (rf/dispatch [:update-theme-and-init-root :shell-stack])
                                (rf/dispatch [:profile/show-testnet-mode-banner-if-enabled])
                                (rf/dispatch [:universal-links/process-stored-event]))
         :type                :primary

--- a/src/status_im/contexts/preview/quo/common.cljs
+++ b/src/status_im/contexts/preview/quo/common.cljs
@@ -25,4 +25,4 @@
                      (rf/dispatch [:navigate-back])
                      (do
                        (rf/dispatch [:theme/switch {:theme :dark}])
-                       (rf/dispatch [:init-root root])))}]))
+                       (rf/dispatch [:update-theme-and-init-root root])))}]))

--- a/src/status_im/contexts/profile/events.cljs
+++ b/src/status_im/contexts/profile/events.cljs
@@ -53,7 +53,7 @@
                     (assoc :profile/profiles-overview profiles)
                     (update :profile/login #(select-profile % key-uid)))
                 db-with-settings)
-          :fx [[:dispatch [:init-root :screen/profile.profiles]]
+          :fx [[:dispatch [:update-theme-and-init-root :screen/profile.profiles]]
                (when (and key-uid userConfirmed)
                  [:effects.biometric/check-if-available
                   {:key-uid    key-uid
@@ -61,7 +61,7 @@
                                  (rf/dispatch [:profile.login/check-biometric-success key-uid
                                                auth-method]))}])]})
        {:db db-with-settings
-        :fx [[:dispatch [:init-root :screen/onboarding.intro]]]}))))
+        :fx [[:dispatch [:update-theme-and-init-root :screen/onboarding.intro]]]}))))
 
 (rf/reg-event-fx
  :profile/update-setting-from-backup

--- a/src/status_im/contexts/profile/login/events.cljs
+++ b/src/status_im/contexts/profile/login/events.cljs
@@ -73,13 +73,13 @@
 
                 (cond
                   pairing-completed?
-                  [[:dispatch [:init-root :screen/onboarding.syncing-results]]]
+                  [[:dispatch [:update-theme-and-init-root :screen/onboarding.syncing-results]]]
 
                   (get db :onboarding/new-account?)
                   [[:dispatch [:onboarding/finalize-setup]]]
 
                   :else
-                  [[:dispatch [:init-root :shell-stack]]
+                  [[:dispatch [:update-theme-and-init-root :shell-stack]]
                    [:dispatch [:profile/show-testnet-mode-banner-if-enabled]]]))})))
 
 ;; login phase 2: we want to load and show chats faster, so we split login into 2 phases
@@ -185,7 +185,7 @@
      {:db (-> db
               (assoc-in [:profile/login :password] password)
               (assoc-in [:profile/login :processing] true))
-      :fx [[:dispatch [:init-root :progress]]
+      :fx [[:dispatch [:update-theme-and-init-root :progress]]
            [:effects.profile/login
             [(get-in db [:profile/login :key-uid])
              (security/safe-unmask-data password)]]]})))

--- a/src/status_im/navigation/core.cljs
+++ b/src/status_im/navigation/core.cljs
@@ -29,7 +29,7 @@
      (re-frame/dispatch [:bottom-sheet-hidden])
      (when @state/root-id
        (reset! theme/device-theme (rn/get-color-scheme))
-       (re-frame/dispatch [:init-root @state/root-id])
+       (re-frame/dispatch [:update-theme-and-init-root @state/root-id])
        (re-frame/dispatch [:chat/check-last-chat])
        (re-frame/dispatch [:alert-banners/restore-alert-banners]))
      (rn/hide-splash-screen)))

--- a/src/status_im/navigation/effects.cljs
+++ b/src/status_im/navigation/effects.cljs
@@ -74,7 +74,6 @@
    (let [[status-bar-theme] (get-status-nav-color root-id theme)
          root               (get (roots/roots status-bar-theme) root-id)]
      (dismiss-all-modals)
-     (rf/dispatch [:theme/switch {:view-id root-id}])
      (reset! state/root-id (or (get-in root [:root :stack :id]) root-id))
      (navigation/set-root root)
      (state/navigation-state-reset [{:id   root-id

--- a/src/status_im/navigation/events.cljs
+++ b/src/status_im/navigation/events.cljs
@@ -70,6 +70,7 @@
     (assoc :dispatch [:chat/close (:current-chat-id db)])))
 
 (rf/defn init-root
+  "WARNING: Use `:update-theme-and-init-root` instead. `:init-root` should not be used directly."
   {:events [:init-root]}
   [{:keys [db]} root-id]
   {:set-root [root-id (:theme db)]})

--- a/src/status_im/navigation/events.cljs
+++ b/src/status_im/navigation/events.cljs
@@ -74,6 +74,12 @@
   [{:keys [db]} root-id]
   {:set-root [root-id (:theme db)]})
 
+(rf/defn update-theme-and-init-root
+  {:events [:update-theme-and-init-root]}
+  [_ root-id]
+  {:fx [[:dispatch [:theme/switch {:view-id root-id}]]
+        [:dispatch [:init-root root-id]]]})
+
 (rf/defn change-tab
   {:events [:navigate-change-tab]}
   [{:keys [db]} stack-id]

